### PR TITLE
Documentation GC

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Update documentation about GC.
- Add ARC based on code observations and https://forum.nim-lang.org/t/5734#35562

Ping @Araq to spot errors, is based on his descriptions.

### CODE REVIEW

**Comment by Araq** [comment]
Will merge it before the next release. Right now ARC is still in heavy development.

File: doc/gc.rst [hunk 1]
```diff
@@ -148,11 +148,11 @@ would be the thread that calls ``dumpNumberOfInstances``.) This might
...
 
-Garbage collector options
-=========================
+Memory Management Strategies
+============================
 
-You can choose which garbage collector to use when compiling source code,
-you can pass ``--gc:`` on the compile command with the choosed garbage collector.
+You can choose the memory management to use when compiling source code,
+you can pass ``--gc:`` on the compile command with the choosed memory management.
```
Comment (ghost) [inline doc/gc.rst]: "choosed" -> "chosen", or maybe even "selected"

File: doc/gc.rst [hunk 2]
```diff
@@ -19,11 +19,8 @@ This document describes how the GC works and how to tune it for
...
 References on the stack are not counted for better performance (and easier C
-code generation). Cycle detection is currently done by a simple mark&sweep
-GC that has to scan the full (thread local heap). ``--gc:v2`` replaces this
-with an incremental mark and sweep. That it is not production ready yet,
-however.
-
+code generation). Cycle detection is currently done by a simple Mark&Sweep
+GC that has to scan the full (thread local heap).
 
 The GC is only triggered in a memory allocation operation. It is not triggered
```
Comment (ghost) [inline doc/gc.rst]: That should also (maybe) be updated, because with --gc:arc that's not really the case

File: doc/gc.rst [hunk 3]
```diff
@@ -149,23 +148,47 @@ would be the thread that calls ``dumpNumberOfInstances``.) This might
...
 
 
-Garbage collector options
--------------------------
+Memory Management Strategies
+============================
 
-You can choose which garbage collector to use when compiling source code,
-you can pass ``--gc:`` on the compile command with the choosed garbage collector.
+You can choose the memory management to use when compiling source code,
```
Comment (ghost) [inline doc/gc.rst]: "memory management option" or "memory management strategy" maybe?

File: doc/gc.rst (hunk 2 above)
Comment (juancarlospaco) [reply doc/gc.rst]: Does it really needs a dedicated introduction section ?.

I think thats all redundant info, and if its not then should be integrated on each garbage collector description, for DRY.
Also it wont apply for ARC nor Manual anyways.

File: doc/gc.rst (hunk 3 above)
Comment (juancarlospaco) [reply doc/gc.rst]: Check again plz

**Pull request merged**